src/app/app.cpp

```diff
@@ -363,6 +363,7 @@ int App::initialize(const AppOptions& options)
 
     // Create the main window.
     m_mainWindow.reset(new MainWindow);
+    m_mainWindow->afterInit();
```

Comment:
During the process of creating the MainWindow, if any error occurs, whatever it is, the Console is called and the constructor creates a connection to the "Close" command of the MainWindow:
```cpp
      // When the main window is closed, we must close the console (in
      // otherwise, the main message loop will continue to run for the duration of the
      // console as well).
      m_mainWindowClosedConn =
        Application::instance()->mainwindow()->Close.connect(
          [this]{ closeWindow(nullptr); });
```
The call to `App::instance()->mainWindow()` results in a crash since no such instance exists yet.
Alternative:
- Add an `if (App::instance()->mainWindow())` condition before creating the connection, and get rid of `afterInit()`. But this produces the following behaviour:
1. The error appears in the aseprite's console.
2. If you want to close Aseprite, it doesn't close, the app waits for the console to close.

What do you think?

Reply:
Thanks for the advice! I implemented something like this but got unexpected results when Aseprite starts with an invalid dithering matrix:
- Sometimes Aseprite starts fine, other times it gives an error

<img width="939" alt="Screenshot 2023-06-26 at 13 08 06" src="https://github.com/aseprite/aseprite/assets/32165475/ebedae86-36d5-403f-81ca-a07568b26d28">
<img width="1300" alt="Screenshot 2023-06-26 at 13 09 49" src="https://github.com/aseprite/aseprite/assets/32165475/ca7ed1e8-9a78-4a11-ad0a-097673501df0">

I can't quite see why this effect occurs. It would take quite a bit more time to get to deeper. The thing is, this doesn't happen if I just move to `initialize()` just the creation and manipulation of the `ContextBar`.

Reply:
I'll try to figure this out today.

Reply:
Ready for review.